backend/src/project/project.resolver.ts

```diff
@@ -0,0 +1,57 @@
+// GraphQL Resolvers for Project APIs
+import {
+    Args,
+    Field,
+    Mutation,
+    ObjectType,
+    Query,
+    Resolver,
+} from '@nestjs/graphql';
+import { ProjectsService } from './project.service';
+import { Projects } from './project.model';
+import { UpsertProjectInput } from './dto/project.input';
+
+@Resolver(() => Projects)
+export class ProjectsResolver {
+  constructor(
+    private readonly projectsService: ProjectsService
+  ) {}
+
+  // -------- All the code need to extract the user id from the token and verify is the project user's or not
+  // add @GetAuthToken() token: string after test
+  @Query(() => [Projects])
+  async getUserProjects(@Args('userId') userId: string): Promise<Projects[]> {
```

Comment:
check the decorator or adding a middleware or adding interceptor before all query get reached

Reply:
lets use a guard and a decorator